core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/internal/CarbonCoreActivator.java

```diff
@@ -69,6 +70,28 @@ public void start(BundleContext context) throws Exception {
                  System.getProperty("user.language") + "-" + System.getProperty("user.country") +
                  ", " + System.getProperty("user.timezone"));
 
+        // Find the wum update level and log it.
+        long updateTimestamp = 0;
+        File updateDirectory = Paths.get(carbonHome, "updates", "wum").toFile();
+        if (updateDirectory.exists() && updateDirectory.isDirectory()) {
+            File[] files = updateDirectory.listFiles();
+            if (files != null) {
+                for (final File file : files) {
+                    if (!file.isDirectory()) {
+                        String fileName = file.getName();
+                        try {
+                            long extractedUpdateTimestamp = Long.parseLong(fileName);
+                            if (extractedUpdateTimestamp > updateTimestamp) {
+                                updateTimestamp = extractedUpdateTimestamp;
+                            }
+                        } catch (NumberFormatException ignore) {
```

Comment:
Why is this exception ignored?

Reply:
There can be hidden or other files where the file names cant parse into a number. Those are get ignored as we cannot do anything about those files.